public/app/features/dashboard-scene/sharing/ShareExportTab.tsx

```diff
@@ -216,21 +217,48 @@ export class ShareExportTab extends SceneObjectBase<ShareExportTabState> impleme
 }
 
 function getMetadata(
-  scene: DashboardScene
+  scene: DashboardScene,
+  isSharingExternally: boolean
 ): DashboardWithAccessInfo<DashboardV2Spec>['metadata'] | Partial<ObjectMeta> {
-  let result = {};
+  let result: Partial<ObjectMeta> = {};
+
   if (scene.serializer.metadata) {
     if ('k8s' in scene.serializer.metadata) {
-      result = scene.serializer.metadata.k8s ?? {};
+      result = scene.serializer.metadata.k8s ? cloneDeep(scene.serializer.metadata.k8s) : {};
     } else if ('annotations' in scene.serializer.metadata) {
-      result = scene.serializer.metadata;
+      result = cloneDeep(scene.serializer.metadata);
     }
   }
 
   if ('managedFields' in result) {
     delete result['managedFields'];
   }
 
+  if (isSharingExternally) {
+    // Remove fields that are not needed for sharing externally
+    if ('uid' in result) {
```

Comment:
why the special check for UID, but not the others?

Reply:
@ryantxu  - we are still using https://github.com/grafana/grafana/blob/schemav2-export-yaml/public/app/features/apiserver/types.ts#L19 as far as I see. We need to update to use dashboard's generate `Metadata` type: https://github.com/grafana/grafana/blob/schemav2-export-yaml/packages/grafana-schema/src/schema/dashboard/v2alpha1/types.metadata.gen.ts#L6